### PR TITLE
Remove constrained view in the 2D editor

### PR DESCRIPTION
**Pull request by Calinou**
We initially [added an option to disable constraining the 2D editor view](https://github.com/godotengine/godot/pull/30041). This setting was still enabled by default to avoid confusing users who end up scrolling too far away from their current scene (which is a problem if you don't know about the <kbd>F</kbd> key to focus on the selection).

However, it's probably a better choice to unconstrain the 2D editor view by default because:

- Lots of people don't know about this setting and wonder how they can scroll far away from the scene. This feels really limiting for them, and it can even lead to some people thinking Godot intentionally limits scene sizes.
- The 3D editor doesn't have such a constrain mechanism. This makes the 2D editor more consistent with the 3D editor.

See also https://github.com/godotengine/godot/issues/25616.

### CODE REVIEW

File: editor/plugins/canvas_item_editor_plugin.cpp
Comment (KoBeWi) [inline editor/plugins/canvas_item_editor_plugin.cpp:4387]: `begin` and `end` should be removed too. Also comment needs update.

**Review by KoBeWi — CHANGES_REQUESTED** [review]

File: editor/plugins/canvas_item_editor_plugin.cpp
Comment (Calinou) [reply editor/plugins/canvas_item_editor_plugin.cpp:4387]: `begin` and `end` are still used below, I can't compile the code if I remove them.

Comment (KoBeWi) [reply editor/plugins/canvas_item_editor_plugin.cpp:4387]: Ah right .-.
But the comment in line 4386 mentions constrain, so it should be changed I guess.

**Review by KoBeWi — APPROVED** [review]
Looks ok. I only ever saw complains (and confusion) about the constrains, so it's good to have them gone. If people won't like this change we can always revert it, but I doubt it will happen.

**Comment by pouleyKetchoupp** [comment]
> This setting was still enabled by default to avoid confusing users who end up scrolling too far away from their current scene (which is a problem if you don't know about the F key to focus on the selection).

Maybe adding a button to focus on the selection, or re-center to the default location could be useful in order to solve the problem of users getting lost far away from the initial location.
I'm not proposing to add it to this PR though, it can be a separate change.

**Comment by groud** [comment]
As @pouleyKetchoupp mentionned, I still don't like the fact that unzooming a lot, then zooming in an empty area may lead to you screen being lost in a completely empty area, with a very wide distance to scroll to get your content back.

I am against removing the constrained view until we find a proper solution to this problem. And I don't think the the 'F' key shortcut is a valid solution here, it's too much of an unknown shortcut, and is not, IMHO, something users would naturally think as a proper proper way to "go back to my 2D editor working state" button.

Maybe an idea would be to add a "center view to content" button (or something like that), in the corner between the scrollbars? Most likely disabled by default, but if you get lost somewhere it gets enabled. That would make it appear as the "solve my problem button" when you are lost.

**Comment by Calinou** [comment]
> Maybe adding a button to focus on the selection, or re-center to the default location could be useful in order to solve the problem of users getting lost far away from the initial location.
I'm not proposing to add it to this PR though, it can be a separate change.

This is done by https://github.com/godotengine/godot/pull/47629.

**Comment by MunWolf** [comment]
My 2 cents on this, I have never found constraining the viewport to be useful.
I usually work in an unrestrained viewport and have never got lost or needed a button to get back.

On that front I don't even think a button to recenter the viewport is necessary as you can always double click a node to center on it in the node list.

The only thing the constrained viewport has done for me is make me annoyed when I reinstall godot and have to go find the option again.

**Comment by YuriSizov** [comment]
> I'm not proposing to add it to this PR though, it can be a separate change.

I agree with that. The concern is valid, but I think we can merge this PR as is, and then if centering the view proves to be unintuitive, I'm sure a PR will be made to fix that. I don't think this change makes the issue worse.

I would argue that viewport constraints are the bigger problem out of two right now. I've been making a trailer for my game last year, fully in engine, and needed to pan a long distance in one scene. The constrained view was very much getting in my way, especially given that it was connected to the zoom level, making it very hard.

@groud Do you insist on blocking this PR until we address the centering UX issue or would you agree on pushing this problem to a follow-up PR?

**Comment by KoBeWi** [comment]
By centering UX you mean the [center view button](https://github.com/godotengine/godot/pull/77632/files), right?

**Comment by YuriSizov** [comment]
Do you have more such useful and on-point PRs up your sleeves? 😆

**Comment by groud** [comment]
I am still not convinced this is better UX, but well, the support for the change seems significant enough for trying it in production, despite my convictions.

So I am ok with merging this PR if we merge Kobewi's PR before.

**Review by YuriSizov — APPROVED** [review]

**Comment by Macksaur** [comment]
Oh this is a shame. I don't want to see this go, I even fixed the weird crusty behaviour in: https://github.com/godotengine/godot/pull/73223

**Pull request merged**
